x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -317,3 +329,50 @@ export const getAgentDataHandler: RequestHandler<
 function isStringArray(arr: unknown | string[]): arr is string[] {
   return Array.isArray(arr) && arr.every((p) => typeof p === 'string');
 }
+
+interface VersionsType {
+  versions: string[];
+}
+
+// Read a static file generated at build time and cache it for subsequent calls
+export const getAvailableVersionsHandler: RequestHandler = async (context, request, response) => {
+  const versionsFile = 'build/kibana/x-pack/plugins/fleet/target/agent_versions_list.json';
```

Comment:
should this be `x-pack/plugins/fleet/target/agent_versions_list.json`?

Reply:
The idea was to have under `build` since it should not be versioned, it's not a typo :)